ttnn/cpp/ttnn/tensor/host_buffer/borrowed_buffer.hpp

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
```

Comment:
no changes, just moved out of host_buffer/types.hpp splitting borrowed_buffer and owned_buffer into their own files